### PR TITLE
make description of security options more clear

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

File: 3.5/security-security-options.md [hunk 1]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
 Several options exists to restrict JavaScript application code functionality 
 to just certain allowed subsets. Which subset of functionality is available
 can be controlled via blacklisting and whitelisting access to individual 
-components. Blacklists can be used to disallow access to dedicated functionality, 
-whereas whitelists can be used to explicitly allow access to certain functionality.
+components.
 
-If an item is covered by both a blacklist and a whitelist, the whitelist will 
-overrule and access to the functionality will be allowed.
+The set thery for these list work as follow:
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
The set theory for these lists works as follow:
```

File: 3.5/security-security-options.md [hunk 2]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+
+- only a blacklist is specifiend: all is allowed except a set of items matching the blacklist.
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
- only a blacklist is specified: all is allowed except the set of items matching the blacklist.
```

Comment (ewoutp) [inline 3.5/security-security-options.md]: typo specifiend

File: 3.5/security-security-options.md [hunk 3]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+- only a whitelist is specified: all is disallowed except a set of items matching the whitelist
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
- only a whitelist is specified: all is disallowed except the set of items matching the whitelist
```

File: 3.5/security-security-options.md [hunk 4]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+- white and blacklist specified: all is disallowed except a set of items mathching the whitelist. From this whitelisted set, subsets can be forbidden again using blacklists.
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
- both whitelist and blacklist are specified: all is disallowed except the set of items matching the whitelist. From this whitelisted set, subsets can be forbidden again using the blacklist.
```

Comment (ewoutp) [inline 3.5/security-security-options.md]: "white and blacklist specified: everything that matches a whitelist item and does not match a blacklist item is allowed."

File: 3.5/security-security-options.md [hunk 5]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
 Values for blacklist and whitelist options need to be specified as ECMAScript 
-regular expressions. Each option can be used multiple times. In this case,
-the individual values for each option will be combined with a _logical or_.
+regular expressions.
+Each option can be used multiple times. When specifying more than one 
+pattern, these patterns will be combined with a _logical or_ to a more 
+complex pattern. 
+
+These patterns and how they are applied can be observed by enabling 
+`--log SECURITY=debug` in the `arangod` or `arangosh` log output.
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
`--log.level SECURITY=debug` in the `arangod` or `arangosh` log output.
```

File: 3.5/security-security-options.md [hunk 6]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+- only a blacklist is specifiend: all is allowed except a set of items matching the blacklist.
+- only a whitelist is specified: all is disallowed except a set of items matching the whitelist
+- white and blacklist specified: all is disallowed except a set of items mathching the whitelist. From this whitelisted set, subsets can be forbidden again using blacklists.
 
 Values for blacklist and whitelist options need to be specified as ECMAScript 
-regular expressions. Each option can be used multiple times. In this case,
-the individual values for each option will be combined with a _logical or_.
+regular expressions.
+Each option can be used multiple times. When specifying more than one 
+pattern, these patterns will be combined with a _logical or_ to a more
```
Comment (ewoutp) [inline 3.5/security-security-options.md]: Don't understand "more complex pattern"
Suggest to leave it out

File: 3.5/security-security-options.md [hunk 7]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+#### Combining patterns
+The pattern to observe most easily is the masquerading of the startup options:
 
     --javascript.startup-options-whitelist "^server\."
     --javascript.startup-options-whitelist "^log\."
     --javascript.startup-options-blacklist "^javascript\."
     --javascript.startup-options-blacklist "endpoint"
 
-will resolve internally to the following regular expressions:
+these sets will resolve internally to the following regular expressions:
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
These sets will resolve internally to the following regular expressions:
```

File: 3.5/security-security-options.md [hunk 8]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+`--log SECURITY=debug` in the `arangod` or `arangosh` log output.
 
-For example, the following combination of startup options
+#### Combining patterns
+The pattern to observe most easily is the masquerading of the startup options:
 
     --javascript.startup-options-whitelist "^server\."
     --javascript.startup-options-whitelist "^log\."
     --javascript.startup-options-blacklist "^javascript\."
     --javascript.startup-options-blacklist "endpoint"
```
Comment (ewoutp) [inline 3.5/security-security-options.md]: Suggest to make it `"^endpoint$"`

File: 3.5/security-security-options.md [hunk 9]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+Each option can be used multiple times. When specifying more than one 
+pattern, these patterns will be combined with a _logical or_ to a more 
+complex pattern. 
+
+These patterns and how they are applied can be observed by enabling 
+`--log SECURITY=debug` in the `arangod` or `arangosh` log output.
 
-For example, the following combination of startup options
+#### Combining patterns
+The pattern to observe most easily is the masquerading of the startup options:
```
Comment (jsteemann) [inline 3.5/security-security-options.md]: I don't understand this sentence :man_shrugging:

File: 3.5/security-security-options.md [hunk 10]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
 
 ```
 --javascript.startup-options-whitelist = "^server\.|^log\."
 --javascript.startup-options-blacklist = "^javascript\.|endpoint"
 ```
 
-Access to directories and files from JavaScript operations is only 
-controlled via a whitelist, which can be specified via the startup
-option `--javascript.files-whitelist`.
+Invoking an arangosh with these options will hide the blacklists from the
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
Invoking an arangosh with these startup options will hide them blacklisted options from the
```

Comment (ewoutp) [inline 3.5/security-security-options.md]: ... will hide the blacklisted items ...

File: 3.5/security-security-options.md [hunk 11]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+output of: 
+
+    require('internal').options()
+
+and an exception will be thrown when trying to access them.
```
Comment (jsteemann) [inline 3.5/security-security-options.md]: Why should there be an exception? Aren't the options just hidden?

File: 3.5/security-security-options.md [hunk 12]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+In contrast to other areas, access to directories and files from JavaScript
+operations is only controlled via a whitelist, which can be specified via the startup
+option `--javascript.files-whitelist`. Thus anything not matching the whitelist
+will be hidden from javascript functions.
 
 For example, when using the following startup options
     
-    --javascript.startup-options-whitelist "^/etc/required/"
-    --javascript.startup-options-whitelist "^/etc/mtab/"
+    --javascript.files-whitelist "^/etc/required/"
```
Comment (ewoutp) [inline 3.5/security-security-options.md]: Now the samples are only about directory.
Would add a file example.

File: 3.5/security-security-options.md [hunk 13]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+output of: 
+
+    require('internal').options()
+
+and an exception will be thrown when trying to access them.
+
+#### File access
+In contrast to other areas, access to directories and files from JavaScript
+operations is only controlled via a whitelist, which can be specified via the startup
+option `--javascript.files-whitelist`. Thus anything not matching the whitelist
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
option `--javascript.files-whitelist`. Thus any files or directories not matching the whitelist
```

File: 3.5/security-security-options.md [hunk 14]
```diff
@@ -40,38 +40,55 @@ an overview of the relevant options.
...
+will be hidden from javascript functions.
```
Comment (jsteemann) [inline 3.5/security-security-options.md]:
```suggestion
will be inaccessible from JavaScript filesystem functions.
```

**Comment by Simran-B** [comment]
@dothebart Are all comments addressed or is there more to tackle?

File: 3.5/security-security-options.md [hunk 15]
```diff
@@ -40,67 +40,134 @@ an overview of the relevant options.
...
   `--javascript.startup-directory`.
 
+#### Endpoint access
+
+The endpoint black/white listing limits access to external HTTP resources. 
+In contrast to the URLs specified in the JavaScript code, the filters have
+to be specified in the ArangoDB endpoints notation: 
+
+- http:// => tcp://
+- https:// => ssl://
```
Comment (Simran-B) [inline 3.5/security-security-options.md]: What about the other aliases we have (at least in client tools)?
`http+tcp`, `http+ssl`, `srv` (Windows only)...

File: 3.5/security-security-options.md [hunk 16]
```diff
@@ -40,67 +40,134 @@ an overview of the relevant options.
...
+ - `https://arangodb.org:777`
+ - `https://arangodb.org`
+ - `http://arangodb.org` 
+ 
+Specifying `ssl://arangodb.org` will match:
+ - `https://arangodb.org:777`
+ - `https://arangodb.org`
+
+Specifying `ssl://arangodb.org:443` will match:
+ - `https://arangodb.org`
```
[the remark this answers is not in the document]
Comment (dothebart) [reply 3.5/security-security-options.md]: https defaults to 443 - so 443 is implicit here.

File: 3.5/security-security-options.md [hunk 17]
```diff
@@ -40,67 +40,134 @@ an overview of the relevant options.
...
 
+#### Endpoint access
+
+The endpoint black/white listing limits access to external HTTP resources. 
+In contrast to the URLs specified in the JavaScript code, the filters have
+to be specified in the ArangoDB endpoints notation: 
+
+- http:// => tcp://
+- https:// => ssl://
+- no protocol will match http and https.
```
[the remark this answers is not in the document]
Comment (dothebart) [reply 3.5/security-security-options.md]: we only know how to do http(s)

File: 3.5/security-security-options.md (hunk 15 above)
Comment (dothebart) [reply 3.5/security-security-options.md]: no, needs to be the propper internal representation as documented. Should I add these to that list?

File: 3.5/security-security-options.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (Simran-B) [reply 3.5/security-security-options.md]: I see, but `https://arangodb.org:443` will also match then...

[the remark this answers is not in the document]
Comment (dothebart) [reply 3.5/security-security-options.md]: exactly

**Comment by Simran-B** [comment]
Merging for now, created issue: https://github.com/arangodb/docs/issues/92

**Pull request merged**